src/components/structures/MessagePanel.js

```diff
@@ -595,6 +595,15 @@ export default class MessagePanel extends React.Component {
 
         const readReceipts = this._readReceiptsByEvent[eventId];
 
+        let isLastSuccessful = false;
```

Comment:
Hmm, this seems to consider events from anyone in the room, not just your own... Is that correct? It might be correct and I'm just having trouble processing it, so good to double-check at least.

Reply:
hmm, it's not intentionally set up this way but ends up working right in the end. Will explore whether documentation or a check is best, as there might be performance penalty? Not entirely sure...